src/Search/SearchResults.php

```diff
@@ -165,7 +166,11 @@ public function __construct(PagerfantaInterface $pagerFantaResults, SearchOption
     private function processResults(): void
     {
         $this->currentPage = $this->searchOptions->getCurrentPage();
-        $this->pagerFantaResults->setCurrentPage($this->currentPage);
+        try {
+            $this->pagerFantaResults->setCurrentPage($this->currentPage);
+        } catch (OutOfRangeCurrentPageException $exception) {
+            $this->pagerFantaResults->setCurrentPage(1);
```

Comment:
Maybe also set `$this-currentPage` to 1. Because this does get serialized back out.
As an alternative you can see how many pages there are, and if it's less than currentpage, then set to 1. You could do that with a ternary if.